api-channel-jira/src/main/java/com/synopsys/integration/alert/api/channel/jira/distribution/search/JiraSearcher.java

```diff
@@ -29,19 +31,24 @@
 import com.synopsys.integration.alert.processor.api.extract.model.project.BomComponentDetails;
 import com.synopsys.integration.alert.processor.api.extract.model.project.ComponentConcernType;
 import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.jira.common.model.response.TransitionsResponseModel;
 
 public abstract class JiraSearcher extends IssueTrackerSearcher<String> {
     private final String jiraProjectKey;
     private final JiraIssueAlertPropertiesManager issuePropertiesManager;
+    private final JiraIssueStatusCreator jiraIssueStatusCreator;
 
-    protected JiraSearcher(String jiraProjectKey, JiraIssueAlertPropertiesManager issuePropertiesManager, ProjectMessageToIssueModelTransformer modelTransformer) {
+    protected JiraSearcher(String jiraProjectKey, JiraIssueAlertPropertiesManager issuePropertiesManager, ProjectMessageToIssueModelTransformer modelTransformer, JiraIssueStatusCreator jiraIssueStatusCreator) {
         super(modelTransformer);
         this.jiraProjectKey = jiraProjectKey;
         this.issuePropertiesManager = issuePropertiesManager;
+        this.jiraIssueStatusCreator = jiraIssueStatusCreator;
     }
 
     protected abstract List<JiraSearcherResponseModel> executeQueryForIssues(String jql) throws IntegrationException;
 
+    protected abstract TransitionsResponseModel fetchIssueTransitions(String issueKey) throws IntegrationException;
```

Comment:
This method doesn't really belong here. It probably deserves its own class (e.g. `JiraIssueTransitionRetriever`).

Reply:
Both Jira Server and Jira Cloud have their own implementations for fetching their transitions. Both will use IssueService but those classes are independent of each other despite having the same name. We could create a JiraServerIssueTransitionRetriever as well as a cloud version but we wouldn't be able to call them from the parent class here without this abstracted method.

Reply:
Let me clarify. Fetching transitions should not be the responsibility of the implementer of `JiraSearcher`. It should be handled by an abstraction that can be passed through the constructor of `JiraSearcher`. I understand that the _implementation_ of that abstraction will be different, but that just means your abstraction should be an interface: 

Example: 
```java
public interface JiraIssueTransitionRetriever {
    TransitionsResponseModel fetchIssueTransitions(String issueKey) throws IntegrationException;
}
```

Reply:
The protected method can now be removed and the place it was called can be replaced with a direct call the the instance of the interface.